### PR TITLE
Refactor (bpdm) - Creation of Auth YML's and Path changes

### PR DESCRIPTION
**Pull request by alexsilva-CGI**
Solves remaining points raised on PR #623.

Solve #572

Should be merged only after #623 conclusion.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [X] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [X] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by nicoprow — CHANGES_REQUESTED** [review]
Please rebase this branch onto the current main, then it will be easier to review

File: bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    gate:
+      oauth2-client-registration: bridge-client
+      security-enabled: true
+
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          catena-keycloak-provider:
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml:32]: This hear specifies a new connection to a oauth2 server named "catena-keycloak-provider". The problem with this is that also the pool-client has the same client name. Better call them separately. Here "gate-oauth2-provider" and in the pool-auth "pool-oauth2-provider". That way both clients don't need to be under the same oauth2 host but may be secured through different servers. (Think about connecting a dev pool with an int gate for testing purposes for example)

File: bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            issuer-uri: ${bpdm.security.permission.auth-server-url:http://localhost:8180}/realms/${bpdm.security.permission.realm:master}
+        registration:
+          bridge-client:
+            authorization-grant-type: client_credentials
+            client-id: ${bpdm.security.permission.client-id}
+            client-secret: ${bpdm.security.credentials.secret}
+            provider: catena-keycloak-provider
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml:39]: Replace the provider name with the new name according to the comment above

File: bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
+
+bpdm:
+  client:
+    gate:
+      oauth2-client-registration: bridge-client
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml:24]: replace bridge-client name with the new one. (See comment further down below)

**Review by nicoprow — CHANGES_REQUESTED** [review]
I gave some feedback exemplary for the Dummy-Bridge configuration. But the same feedback counts for all the other applications as well.

File: bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  security:
+    oauth2:
+      client:
+        provider:
+          gate-oauth2-provider:
+            issuer-uri: ${bpdm.security.auth-server-url:http://localhost:8180}/realms/${bpdm.security.realm:master}
+        registration:
+          gate-client:
+            authorization-grant-type: client_credentials
+            client-id: ${bpdm.security.permission.client-id}
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-gate-auth.yml:37]: Should by now bpdm.security.client-id as it is not under permission anymore

File: bpdm-bridge-dummy/src/main/resources/application-pool-auth.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      oauth2-client-registration: pool-client
+      security-enabled: true
+
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          pool-oauth2-provider:
+            issuer-uri: ${bpdm.security.permission.auth-server-url:http://localhost:8180}/realms/${bpdm.security.permission.realm:master}
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-pool-auth.yml:32]: Also contains incorrectly 'permission'

File: bpdm-bridge-dummy/src/main/resources/application-pool-auth.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        registration:
+          pool-client:
+            authorization-grant-type: client_credentials
+            client-id: ${bpdm.security.permission.client-id}
```
Comment (nicoprow) [inline bpdm-bridge-dummy/src/main/resources/application-pool-auth.yml:36]: Also here

File: bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/config/SecurityConfig.kt [hunk 7]
```diff
@@ -54,7 +54,7 @@ class NoAuthenticationConfig {
 
 @EnableWebSecurity
 @ConditionalOnProperty(
-    value = ["bpdm.security.enabled"],
+    value = ["bpdm.security.permission.enabled"],
```
Comment (nicoprow) [inline bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/config/SecurityConfig.kt:57]: Should be without permission

File: bpdm-gate/src/main/resources/application-auth.yml [hunk 8]
```diff
@@ -41,14 +41,15 @@ bpdm:
...
 #Spring OAuth configuration
 spring:
   security:
     oauth2:
       resourceserver:
         jwt:
-          issuer-uri: ${bpdm.security.auth-server-url}/realms/${bpdm.security.realm}
-          jwk-set-uri: ${bpdm.security.auth-server-url}/realms/${bpdm.security.realm}/protocol/openid-connect/certs
+          issuer-uri: ${bpdm.security.permission.auth-server-url}/realms/${bpdm.security.permission.realm}
+          jwk-set-uri: ${bpdm.security.permission.auth-server-url}/realms/${bpdm.security.permission.realm}/protocol/openid-connect/certs
```
Comment (nicoprow) [inline bpdm-gate/src/main/resources/application-auth.yml:52]: permission in path here as well

File: bpdm-gate/src/main/resources/application-orchestrator-auth.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  orchestrator-security:
+    oauth2-client-registration: orchestrator-client
+
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          orchestrator-oauth2-provider:
+            issuer-uri: ${bpdm.security.permission.auth-server-url:http://localhost:8180}/realms/${bpdm.security.permission.realm:master}
```
Comment (nicoprow) [inline bpdm-gate/src/main/resources/application-orchestrator-auth.yml:33]: here

File: bpdm-gate/src/main/resources/application-orchestrator-auth.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        registration: # Note that the oauth2-client-registration property is NOT the client id of the gate in keycloak
+          orchestrator-client:
+            authorization-grant-type: client_credentials
+            client-id: ${bpdm.security.permission.client-id}
```
Comment (nicoprow) [inline bpdm-gate/src/main/resources/application-orchestrator-auth.yml:37]: here

File: bpdm-pool/src/main/resources/application-orchestrator-auth.yml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    orchestrator:
+      security-enabled: true
+
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          orchestrator-oauth2-provider:
+            issuer-uri: ${bpdm.security.permission.auth-server-url:http://localhost:8180}/realms/${bpdm.security.permission.realm:master}
```
Comment (nicoprow) [inline bpdm-pool/src/main/resources/application-orchestrator-auth.yml:31]: here

File: bpdm-pool/src/main/resources/application-orchestrator-auth.yml [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        registration: # Note that the oauth2-client-registration property is NOT the client id of the gate in keycloak
+          gate-client:
+            authorization-grant-type: client_credentials
+            client-id: ${bpdm.security.permission.client-id}
```
Comment (nicoprow) [inline bpdm-pool/src/main/resources/application-orchestrator-auth.yml:35]: and here

**Review by nicoprow — CHANGES_REQUESTED** [review]
Looks good, just contains some incorrect path names with 'permission'. Since the properties are now again under security, those paths have to be adjusted

**Comment by alexsilva-CGI** [comment]
Hello @nicoprow, I believe I did the required chances. Is there anything missing? Thanks

**Review by nicoprow — APPROVED** [review]
Everything looks good, I just wanted to wait for the merge after the latest alpha release

**Pull request merged**